common/src/main/java/org/jboss/hal/testsuite/creaper/command/BackupAndRestoreAttributes.java

```diff
@@ -0,0 +1,163 @@
+            }
+        }
+
+        private void addToBatch(Property node) {
+            batch.writeAttribute(address,
+                    node.getName(),
+                    node.getValue()); //add dependent attribute to batch
+            processedAttributes.add(node.getName()); //add to processed attributes
+        }
+
+        private void addAllToBatch(Collection<Property> nodes) {
+            nodes.forEach(this::addToBatch);
+        }
+
+        private void addAllWaitingDependenciesToBatch(Property dependency) {
+            String dependencyName = dependency.getName();
+            if (waitingForDependency.containsKey(dependencyName)) { //attribute which was added to batch is dependency for some other attribute
+                addAllToBatch(waitingForDependency.get(dependencyName)); //add dependent attributes to batch
+                waitingForDependency.remove(dependencyName);
+            }
+        }
+
+        @Override
+        public void apply(OnlineCommandContext ctx) throws Exception {
+            if (BackupAndRestoreAttributes.this.backup == null) {
+                throw new CommandFailedException("There is no backup to be restored!");
+            }
+
+            for (Property processingAttribute : BackupAndRestoreAttributes.this.backup.asPropertyList()) {
+                String name = processingAttribute.getName();
+
+                if (excluded == null || !excluded.contains(name)) { //attribute is not excluded
+                    if (dependencies != null && dependencies.containsKey(name)) { //processed element depends on something
+                        if (processedAttributes.contains(dependencies.get(name))) { //dependency is already processed in batch
+                            addToBatch(processingAttribute);
+                        } else {
+                            putToWaitingQueue(dependencies.get(name), processingAttribute);
+                        }
+                    } else {
+                        addToBatch(processingAttribute);
+                        addAllWaitingDependenciesToBatch(processingAttribute);
+                    }
+                }
+            }
+
+            Operations ops = new Operations(ctx.client);
+            ops.batch(batch);
+        }
+
+    };
+
+    public static final class Builder {
+
+        private Address address;
+        private Map<String, String> dependencies;
+        private Set<String> excluded;
+
+        public Builder(Address address) {
+            this.address = address;
+        }
+
+        /**
+         * Add dependency
+         */
+        public Builder dependency(String attribute, String dependsOn) {
+            if (dependencies == null && dependsOn != null) {
+                dependencies = new HashMap<>();
+            }
+            if (dependencies != null) {
+                dependencies.put(attribute, dependsOn);
+            }
+            return this;
+        }
```

Comment:
attribute might depend only on single attribute? Isn't it possible to depend on more?